packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -383,6 +383,88 @@ it('should handle row clicks', async () => {
+  expect(onSelect).toHaveBeenCalledTimes(3 + (3 + 3)); // # times before + (to reset all depth 0 rows + # selected rows from start to end)
+  expect(onRowClick).toHaveBeenCalledTimes(4);
+});
+
+it('should handle sub-rows shift click selection', async () => {
+  const onSelect = jest.fn();
+  const onRowClick = jest.fn();
+  const data = mockedSubRowsData();
+  const { container, getByText } = renderComponent({
+    data,
+    onSelect,
+    onRowClick,
+    isSelectable: true,
+  });
+
+  let rows = container.querySelectorAll('.iui-table-body .iui-row');
+  expect(rows.length).toBe(3);
+
+  const user = userEvent.setup();
+  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
+  await user.click(getByText(data[1].name)); // [shiftKey: true]
+
+  // By default, when no row is selected before shift click, start selecting from first row to clicked row
+  expect(rows[0].classList).toContain('iui-selected');
+  expect(rows[1].classList).toContain('iui-selected');
+  expect(rows[2].classList).not.toContain('iui-selected');
+  expect(onSelect).toHaveBeenCalledTimes(3 + 2); // to reset all depth 0 rows + # selected rows from start to end
+  expect(onRowClick).toHaveBeenCalledTimes(1);
+
+  await expandAll(container);
+
+  rows = container.querySelectorAll('.iui-table-body .iui-row');
+  expect(rows.length).toBe(10);
+
+  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
+  await user.click(getByText(data[0].subRows[1].subRows[0].name)); // [shiftKey: true]
+
+  let checkboxes = container.querySelectorAll<HTMLInputElement>(
+    '.iui-table-body .iui-checkbox',
+  );
+  let selectedIndices = [1, 3];
+  Array.from(checkboxes).forEach((checkbox, index) => {
+    expect(!!checkbox.checked).toBe(selectedIndices.includes(index));
+  });
+
+  expect(onSelect).toHaveBeenCalledTimes(5 + (3 + 2)); // # times before + (to reset all depth 0 rows + # selected rows from start to end)
+  expect(onRowClick).toHaveBeenCalledTimes(2);
+
+  rows = container.querySelectorAll('.iui-table-body .iui-row');
+  expect(rows.length).toBe(10);
+
+  await user.keyboard('[/ShiftLeft]'); // Release Shift
+  await user.click(getByText(data[0].subRows[1].subRows[1].name)); // [shiftKey = false]
+
+  selectedIndices = [4];
+  Array.from(checkboxes).forEach((checkbox, index) => {
+    expect(!!checkbox.checked).toBe(selectedIndices.includes(index));
+  });
+
+  expect(onSelect).toHaveBeenCalledTimes(10 + 1); // # times before + 1 selection
+  expect(onRowClick).toHaveBeenCalledTimes(3);
+
+  await user.keyboard('[ShiftLeft>]'); // Press Shift (without releasing it)
+  await user.click(getByText(data[1].subRows[0].name)); // [shiftKey = true]
+
+  checkboxes = container.querySelectorAll<HTMLInputElement>(
+    '.iui-table-body .iui-checkbox',
+  );
+  selectedIndices = [4, 5, 7];
+  Array.from(checkboxes).forEach((checkbox, index) => {
+    expect(!!checkbox.checked).toBe(selectedIndices.includes(index));
+  });
```

Comment:
I am not sure, but is `!!` really needed? Can it just be `checkbox.checked`. I kept it in since I saw it on line 1546.

Reply:
`!!` is usually needed to force a value to be boolean (from a different type, e.g. a string). If `checked` is a boolean, it shouldn't be needed

Reply:
Ohh, that's a good concept. But here and on line 1546, I think `checkbox.checked` is a boolean

Reply:
@mayank99 When you get time, can you please let me know if `!!` can be omitted since `checkbox.checked` is already a boolean? Thanks. I know you already said it can be omitted:

> If `checked` is a boolean, it shouldn't be needed

Also, removing `!!` in all places still passes the tests. But since the old code still had `!!`, I just want to confirm if `!!` is needed or not.